packages/angular/common/src/directives/control-value-accessors/radio-value-accessor.ts

```diff
@@ -16,6 +16,21 @@ export class RadioValueAccessorBaseDirective extends ValueAccessor {
     super(injector, el);
   }
 
+  override ngAfterViewInit() {
+    super.ngAfterViewInit();
+    this.handleRadioGroupValueChange()
+  }
+
+  private handleRadioGroupValueChange(): void {
+    const parentElement = this.elementRef.nativeElement.parentElement;
+    if (parentElement) {
```

Comment:
We need to make sure this is the expected parent not some div container in between.

1. Check for element type
2. Traverse up the DOM until correct element was encountered